src/diversity/__main__.py

```diff
@@ -36,48 +35,17 @@ def main(args):
     LOGGER.info(" ".join([f"python{python_version()}", *argv]))
     LOGGER.debug(f"args: {args}")
 
-    delimiter = get_file_delimiter(args.input_filepath)
-    species_counts = read_csv(
-        args.input_filepath,
-        sep=delimiter,
-        usecols=[args.subcommunity_column, args.species_column, args.count_column],
-        dtype={
-            args.subcommunity_column: str,
-            args.species_column: str,
-            args.count_column: int,
-        },
-    )
-
-    LOGGER.debug(f"data: {species_counts}")
-
-    if args.subcommunities is None:
-        subcommunities = unique(species_counts[args.subcommunity_column])
-    else:
-        subcommunities = array(args.subcommunities)
-
-    meta = make_metacommunity(
-        counts=species_counts,
-        subcommunities=subcommunities,
-        similarity_method=args.similarity_matrix_filepath,
-        subcommunity_column=args.subcommunity_column,
-        species_column=args.species_column,
-        count_column=args.count_column,
-        similarity_kwargs={"chunk_size": args.chunk_size},
-    )
-
-    community_views = []
-    for view in args.viewpoint:
-        community_views.append(meta.subcommunities_to_dataframe(view))
-        community_views.append(meta.metacommunity_to_dataframe(view))
-
-    community_views = concat(community_views, ignore_index=True)
-    community_views.viewpoint = community_views.viewpoint.map(
-        lambda v: format(v, ".2f")
+    counts = read_csv(args.input_filepath, sep=None, engine="python", dtype=int64)
```

Comment:
How safe is sep=None?

Reply:
I believe `pandas.read_csv` samples from the file and uses `csv.Sniffer` from  the standard library. This approach feels safer than relying on the user to give the correct extension. It also allows for other delimiters (e.g., ";").